flocker/node/agents/test/test_blockdevice.py

```diff
@@ -49,6 +53,37 @@
     skip = "flocker.node.agents.blockdevice is only supported on Linux"
 
 
+def mkfs(device):
+    """
+    Synchronously initialize a device file with an ext4 filesystem.
+
+    :param FilePath device: The path to the file onto which to put the
+        filesystem.  Anything accepted by ``mkfs`` is acceptable (including a
+        regular file instead of a device file).
+    """
+    options = []
+    if not device.isBlockDevice():
+        options.extend([
+            # Make the filesystem even if the destination is not a block
+            # device.
+            b"-F",
```

Comment:
Actually it looks like there isn't a long option :frowning: but the man page also says

```
       -F     Force mke2fs to create a filesystem, even if the specified device is not a partition on a block special device, or if other  parame‐
              ters  do  not make sense.  In order to force mke2fs to create a filesystem even if the filesystem appears to be in use or is mounted
              (a truly dangerous thing to do), this option must be specified twice.
```

...which makes me wonder why this isn't also needed when we attempt to format an unpartitioned loopback device. Perhaps if you attempted to do this on a "disk like" block device it'd check for a master boot record or something?